providers/implementations/storemgmt/file_store.c

```diff
@@ -450,13 +450,11 @@ static int file_setup_decoders(struct file_ctx_st *ctx)
              * attached to any provider, since it's only used locally.
              */
             to_obj = ossl_decoder_from_algorithm(0, to_algo, NULL);
-            if (to_obj == NULL)
+            if (to_obj != NULL)
+                to_obj_inst = ossl_decoder_instance_new(to_obj, ctx->provctx);
+            OSSL_DECODER_free(to_obj);
```

Comment:
I would put this inside the if() block above.

Reply:
As you have an if for `to_obj != NULL` there is no point in calling the free if to_obj is NULL. But of course this is just a style issue.

Reply:
Not having it in the `if` body makes this another place that exercises the check for NULL that any freeing function is required to do as per our (well, not just our, really) standards.  I exercise that as much as I can, reasoning that one more spot is better than no spot at all.

Reply:
OK, that's valid reasoning as well.